src/IECoreMaya/SceneShapeSubSceneOverride.cpp

```diff
@@ -0,0 +1,1615 @@
+	RenderItemMap renderItems;
+	visitSceneLocations( m_sceneInterface, renderItems, container, Imath::M44d(), /* isRoot = */ true );
+
+}
+
+bool SceneShapeSubSceneOverride::getInstancedSelectionPath( const MRenderItem &renderItem, const MIntersection &intersection, MDagPath &dagPath ) const
+{
+	auto it = m_renderItemNameToDagPath.find( std::string( renderItem.name().asChar() ) );
+	if( it != m_renderItemNameToDagPath.end() )
+	{
+		dagPath.set( it->second );
+		return true;
+	}
+
+	return false;
+}
+
+void SceneShapeSubSceneOverride::updateSelectionGranularity( const MDagPath &path, MSelectionContext &selectionContext )
+{
+	MDagPath parent( path );
+	parent.pop();
+
+	if( componentsSelectable( parent ) )
+	{
+		selectionContext.setSelectionLevel( MHWRender::MSelectionContext::kComponent );
+	}
+	else
+	{
+		selectionContext.setSelectionLevel( MHWRender::MSelectionContext::kObject );
+	}
+}
+
+DrawAPI SceneShapeSubSceneOverride::supportedDrawAPIs() const
+{
+	return kAllDevices;
+}
+
+SceneShapeSubSceneOverride::SceneShapeSubSceneOverride( const MObject& obj )
+	: MPxSubSceneOverride( obj ), m_drawTagsFilter( "" ), m_time( -1 ), m_drawRootBounds( false ), m_shaderOutPlug(), m_materialIsDirty( true ), m_instancedRendering( false /* instancedRendering switch */ ), m_geometryVisible( false )
+{
+	MStatus status;
+	MFnDependencyNode node( obj, &status );
+
+	if( status )
+	{
+		m_sceneShape = dynamic_cast<SceneShape*>( node.userNode() );
+	}
+}
+
+void SceneShapeSubSceneOverride::visitSceneLocations( ConstSceneInterfacePtr sceneInterface, RenderItemMap &renderItems, MSubSceneContainer &container, const Imath::M44d &matrix, bool isRoot )
+{
+	if( !sceneInterface )
+	{
+		return;
+	}
+
+	// TODO: switch this back to accumulating transformation while walking the tree if we find that this gets too expensive
+	Imath::M44d accumulatedMatrix = worldTransform( sceneInterface.get(), m_time );
+	MMatrix mayaMatrix = IECore::convert<MMatrix, Imath::M44d>( accumulatedMatrix );
+
+	// Dispatch to all children.
+	SceneInterface::NameList childNames;
+	sceneInterface->childNames( childNames );
+
+	for( const auto &childName : childNames )
+	{
+		visitSceneLocations( sceneInterface->child( childName ), renderItems, container, accumulatedMatrix );
+	}
+
+	// Now handle current location.
+
+	std::string name;
+	IECoreScene::SceneInterface::Path path;
+	sceneInterface->path( path );
+	IECoreScene::SceneInterface::pathToString( path, name );
+
+	if( isRoot && m_drawRootBounds )
+	{
+		std::string rootItemName = name + "_root";
+
+		// \todo: There's gotta be a better way...
+		Imath::Box3d boundingBoxD = sceneInterface->readBound( m_time );
+		Imath::Box3f boundingBox( Imath::V3f( boundingBoxD.min.x, boundingBoxD.min.y, boundingBoxD.min.z ), Imath::V3f( boundingBoxD.max.x, boundingBoxD.max.y, boundingBoxD.max.z ) );
+		const MBoundingBox mayaBoundingBox = IECore::convert<MBoundingBox>( boundingBox ); //\todo: can this be constructed from Box3d?
```

Comment:
The "IECoreMaya/Convert.h" header suggests it can - is that not working?

Reply:
Ah yes, it is of course. I think that comment slipped through and my main gripe was with converting between Box3f and Box3d. I've addressed all of this in d48d761 and the double->float conversion happens down the road now.